railties/lib/rails/commands/server.rb

```diff
@@ -94,7 +94,8 @@ def default_options
         environment:        (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development").dup,
         daemonize:          false,
         caching:            nil,
-        pid:                Options::DEFAULT_PID_PATH
+        pid:                Options::DEFAULT_PID_PATH,
+        restart_cmd:        restart_command
```

Comment:
By "work" I mean that they won't blow up on unsupported keys for instance (not that they will automatically use the generated command).